src/Zigzag_persistence/include/gudhi/Zigzag_persistence.h

```diff
@@ -0,0 +1,751 @@
+   * assumed to end at the same time than the simplices range and has the same order. The filtration values should be 
+   * ascending in this order and they are assumed to be larger or equal to previously used filtration values.
+   */
+  template <class SimplexRangeIterators, class FiltrationRangeIterators>
+  void remove_simplices_contiguously(SimplexRangeIterators simplex_range_start, SimplexRangeIterators simplex_range_end,
+                                     FiltrationRangeIterators filtration_range_start) {
+    for (; simplex_range_start != simplex_range_end; ++simplex_range_start, ++filtration_range_start) {
+      remove_simplex(*simplex_range_start, *filtration_range_start);
+    }
+  }
+
+  /**
+   * @brief Returns the ``index persistence diagram'' of the current filtration, that is, the pairs of atomic arrow 
+   * numbers corresponding to a birth-death pair. Does not contain points at infinity, only the cycle classes which 
+   * already died are represented.
+   *
+   * @return Reference to the list of intervals.
+   */
+  const std::list<index_interval>& get_index_persistence_diagram() const { return persistence_diagram_; }
+
+  /**
+   * @brief Returns the filtration values \f$[f(b),f(d)]\f$ associated to the indices \f$[b,d]\f$ which are retrieved
+   * by @ref get_index_persistence_diagram.
+   *
+   * @param b_key Birth index
+   * @param d_key Death index
+   * @return A pair of filtration values associated to the given indices.
+   */
+  std::pair<Filtration_value, Filtration_value> map_index_to_filtration_value(
+    Simplex_key b_key, Simplex_key d_key) const 
+  {
+    // filtration_values_ must be sorted by increasing keys.
+    auto it_b =  // lower_bound(x) returns leftmost y s.t. x <= y
+        std::lower_bound(
+            filtration_values_.begin(), filtration_values_.end(),
+            std::pair<Simplex_key, Filtration_value>(b_key, std::numeric_limits<Filtration_value>::infinity()),
+            [](std::pair<Simplex_key, Filtration_value> p1, std::pair<Simplex_key, Filtration_value> p2) {
+              return p1.first < p2.first;
+            });
+    if (it_b == filtration_values_.end() || it_b->first > b_key) {
+      --it_b;
+    }
+    // it points to the rightmost z such that z <= x
+
+    auto it_d =  //
+        std::lower_bound(
+            filtration_values_.begin(), filtration_values_.end(),
+            std::pair<Simplex_key, Filtration_value>(d_key, std::numeric_limits<Filtration_value>::infinity()),
+            [](std::pair<Simplex_key, Filtration_value> p1, std::pair<Simplex_key, Filtration_value> p2) {
+              return p1.first < p2.first;
+            });
+    if (it_d == filtration_values_.end() || it_d->first > d_key) {
+      --it_d;
+    }
+
+    return std::make_pair(it_b->second, it_d->second);
+  }
+
+  /**
+   * @brief Returns the current persistence diagram ordered first by length, than by dimension, 
```

Comment:
Is it really the role of this function to sort the intervals? The cost is probably negligible compared to the rest, but if I don't care about the order or if I want a different order, I am not very happy paying for this useless sort. Actually, it looks like it calls `_get_persistence_diagram` which also sorts with a different order, so we always pay for 2 sorts?

Reply:
We don't have to sort them. I think it just helped me for the debugging (or it was like that in Clement's original interface).